library/src/main/java/com/novoda/downloadmanager/lib/Clock.java

```diff
@@ -0,0 +1,36 @@
+package com.novoda.downloadmanager.lib;
+
+import android.os.SystemClock;
+
+import java.util.concurrent.TimeUnit;
+
+class Clock {
+
+    public enum Interval {
```

Comment:
No I haven't but the problem in this case is the database query, not the code by itself.

I mean for example
```java
if (controlStatus.isPaused())
```

Would be slower than
```java
if (controlStatus.isPaused)
```

But those kinds of micro-optimisations are not really necessary here for now :) The db is the big problem :)